File: uoo6uoow/Week12/큐.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(reader.readLine());

        Deque<Integer> queue = new ArrayDeque<>();

        for (int i = 0; i < n; i++) {
            String input = reader.readLine();
            if (input.equals("front")) {
                if (!queue.isEmpty()){
                    writer.write(queue.peek() + "\n");
                }else {
                    writer.write("-1\n");
                }
            } else if (input.equals("back")) {
                if (!queue.isEmpty()){
                    writer.write(queue.getLast() + "\n");
                }else {
                    writer.write("-1\n");
                }
            } else if (input.equals("size")) {
                writer.write(queue.size() + "\n");

            } else if (input.equals("empty")) {
                if (queue.isEmpty()){
                    writer.write("1\n");
                }else
                    writer.write("0\n");

            } else if (input.equals("pop")) {
                if (!queue.isEmpty()){
                    writer.write(queue.poll() + "\n");
                }else {
                    writer.write("-1\n");
                }
            }else if (input.startsWith("push")){
                String[] com = input.split(" ");
                int num = Integer.parseInt(com[1]);
                queue.offer(num);
            }
        }
        writer.flush();
    }
}

File: noeyeyh/Week12/큐.py
from collections import deque
import sys
input = sys.stdin.readline

n = int(input())
q = deque()
size = 0

for i in range(n):
    l = list(input().split())

    if l[0] == "push":
        q.append(l[1])
        size += 1

    elif l[0] == "pop":
        if size != 0:
            print(q.popleft())
            size -= 1
        else:
            print(-1)

    elif l[0] == "size":
        print(size)

    elif l[0] == "empty":
        if size:
            print(0)
        else:
            print(1)

    elif l[0] == "front":
        if size != 0:
            print(q[0])
        else:
            print(-1)

    else:
        if size != 0:
            print(q[-1])
        else:
            print(-1)